src/EcologicalNetworks.jl

```diff
@@ -91,15 +91,31 @@ export null1, null2, null3, null4
 include(joinpath(".", "rand/RDPG.jl"))
 export RDPG
 
-# Random networks from structural models
-include(joinpath(".", "structuralmodels/cascademodel.jl"))
-export cascademodel
-include(joinpath(".", "structuralmodels/mpnmodel.jl"))
-export mpnmodel
-include(joinpath(".", "structuralmodels/nestedhierarchymodel.jl"))
-export nestedhierarchymodel
-include(joinpath(".", "structuralmodels/nichemodel.jl"))
-export nichemodel
+# Random networks from generative models
+include(joinpath(".", "generativemodels/_generators.jl"))
+export NetworkGenerator, rand, _sizeof_uv_getaddrinfo
+
+include(joinpath(".", "generativemodels/cascademodel.jl"))
+export CascadeModel
+include(joinpath(".", "generativemodels/mpnmodel.jl"))
+export MinimumPotentialNicheModel
+include(joinpath(".", "generativemodels/nestedhierarchymodel.jl"))
+export NestedHierarchyModel
+include(joinpath(".", "generativemodels/nichemodel.jl"))
+export NicheModel
+include(joinpath(".", "generativemodels/erdosrenyi.jl"))
+export ErdosRenyi
+include(joinpath(".", "generativemodels/blockmodel.jl"))
+export BlockModel
+include(joinpath(".", "generativemodels/configurationmodel.jl"))
+export ConfigurationModel
+include(joinpath(".", "generativemodels/preferentialattachment.jl"))
+export PreferentialAttachment
```

Comment:
What do you think about not exporting any of them and dropping `Model` from the end? This would make them slightly less redundant?

Reply:
so "generativemodels/" -> "generative" and drop model from all the type and file names?

Reply:
alternatively all these models could go in `rand/`

Reply:
just pushed name changes @tpoisot

Reply:
also a performance concern i noticed is checking types in `_generate` is a minor slowdown and could be avoided via immutable structs with parameter checks in constructors, perhaps for a new pr